photon-lib/src/test/scala/com/linkedin/photon/ml/optimization/LBFGSBTest.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.optimization
+
+import breeze.linalg.DenseVector
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.normalization.{NoNormalization, NormalizationContext}
+import com.linkedin.photon.ml.test.{Assertions, CommonTestUtils}
+import org.apache.spark.broadcast.Broadcast
+import org.mockito.Mockito.{doReturn, mock}
+import org.testng.Assert.assertEquals
+import org.testng.annotations.{DataProvider, Test}
+
+class LBFGSBTest {
+
+  // TODO: set up boundary with non-zero expected loss value.
+  @DataProvider(name = "dataProvider")
+  def dataProvider(): Array[Array[Any]] = {
+    Array(
+      // Note that expected value here is the value of the function at the unconstrained optima since the
+      // projection happens after it
+      Array(Array(Double.NegativeInfinity), Array(Double.PositiveInfinity), Array(4.0), 0.0),
+      Array(Array(-5.0), Array(5.0), Array(4.0), 0.0),
+      // set to -neg to 5.0 has convergence issue.
+      Array(Array(Double.NegativeInfinity), Array(3.0), Array(3.0), 1.0),
+      Array(Array(-1.0), Array(Double.PositiveInfinity), Array(4.0), 0.0)
```

Comment:
This case doesn't test anything new and should be replaced by the [5, inf) test case.

Reply:
I want to pass this unit test but results in the optimizer error:
`Array(Array(5.0), Array(Double.PositiveInfinity), Array(5.0), 1.0)`
The reason is in breeze.LBFGSB, it will require the initial coeffs in the [lower_bound, upper_bound] range. However, the initial coeffs are generated with mean of 0 and std of 1 as in here:
`val initialCoefficients = CommonTestUtils.generateDenseVector(expectedCoef.length)`
It could be possible to generate the initialCoefficients with mean of (lower_bound+upper_bound)/2, however, it does not support Double.Infinity. 
`(Double.PositiveInfinity + 100.0)/2  //will be Infinity in scala`
One way is passing a really large value instead of Double.Infinity. For example:
`Array(Array(5.0), Array(1000.0), Array(5.0), 1.0)`